### DISCUSSION TITLE
Support for /arch:AVX512

### DISCUSSION THREAD

**Discussion by dgregorius** [premake/premake-core#2106] [General]
Any plans to add this? Or is there another way to set this in the current version?

Thanks!

**Comment by nickclark2016** [comment]
There is currently no support for AVX512 in Premake, but it would be fairly easy to add.  AVX512 is simple enough for Visual Studio, as it's only one flag. GCC and Clang have several flags for AVX512 generation.

**Comment by nickclark2016** [comment]
Just to follow up, in it's current state, Premake allows you to pass options to the compiler via `buildoptions`, so you can pass the appropriate flags.  For example, you could do:
```lua
buildoptions {
  "-mavx512f",
  "-mavx512pf",
  "-mavx512er",
  "-mavx512cd",
  "-mavx512vl",
  "-mavx512bw",
  "-mavx512dq",
  "-mavx512ifma",
  "-mavx512vbmi"
} 
```
And those flags would work for Clang and GCC.  Clang also has a couple others you can use.

For VS, you'd use:
```lua
buildoptions { "/arch:AVX512" }
```

**Comment by dgregorius** [comment]
Thanks Nick!

I tried the buildoptions{} as you are suggesting and unfortunately this seems not to work. It seems the VS project files are not picking it up. I checked both the property grid entry for the AVX settings and used commandline. 

This is on VS2022 Community on Windows 10 Professional.

**Reply by nickclark2016** [reply]
Build Options will not place it there, but with the `AdditionalOptions` entry in the project file.
